# ★Chapter4　Python基础入门

- **4.1 Python基础**
    - **4.1.1 Hello world**
    - **4.1.2 Python的用途**
    - **4.1.3 注释的输入**
    - **4.1.4 数值与字符串**
    - **4.1.5 运算**
<br><br>
- **4.2 变量**
    - **4.2.1 变量的定义** 
    - **4.2.2 变量的更新**
    - **4.2.3 字符串的连接**
<br><br>
- **4.3 类型**
    - **4.3.1 类型的种类**
    - **4.3.2 类型的变换**
    - **4.3.3 对类型的理解与确认**
<br><br>
- **4.4 if文**
    - **4.4.1 比较运算符**
    - **4.4.2 if文**
    - **4.4.3 else**
    - **4.4.4 elif**
    - **4.4.5 and・not・or**
<br><br>
- **附加习题**

## ●4.1 Python基础

### ○4.1.1 Hello world

#### □习题

In [ ]:
# 在 Python 当中，如果一行的开头使用了 # 就会被作为注释处理
# 请在屏幕上输出“Hello world”
print("Hello world")

**程序清单 4.1：习题**

#### □参考答案

In [1]:
# 在 Python 当中，如果一行的开头使用了 # 就会被作为注释处理
# 请在屏幕上输出“Hello world”
print("Hello world")

Hello world


**程序清单 4.2：参考答案**

### ○4.1.2  Python的用途

这里没有样本。

### ○4.1.3 注释的输入

#### □习题

In [ ]:
# 请输出5+2的结果
print(5 + 2)


print(3 + 8)


**程序清单4.3：习题**

#### □参考答案

In [3]:
# 请输出5+2的结果
print(5 + 2)

# 请输出3+8的结果
print(3 + 8)

7
11


**程序清单4.4：参考答案**

### ○4.1.4 数值与字符串

In [4]:
print(3 + 6)

9


**程序清单 4.5：计算的例子①**

In [5]:
print("8 - 3")

8 - 3


**程序清单4.6：计算的例子②**

#### □习题

In [ ]:
# 请输出数值18


# 请输出数值2+6的结果


# 请输出“2+6”的字符串


**程序清单4.7：习题**

#### □参考答案

In [3]:
# 请输出数值18
print(18)

# 请输出数值2+6的结果
print(2 + 6)

# 请输出“2+6”的字符串
print("2 + 6")

18
8
2 + 6


**程序清单4.8：参考答案**取程序清单

### ○4.1.5 运算

#### □习题

In [ ]:
# 3 + 5


# 3 - 5


# 3 × 5


# 3 ÷ 5


# 3除以5得到的余数


# 3和5的幂乘



**程序清单4.9：习题**

#### □参考答案

In [4]:
# 3 + 5
print(3 + 5)

# 3 - 5
print(3 - 5)

# 3 × 5
print(3 * 5)

#　3 ÷ 5
print(3 / 5)

# 3除以5得到的余数
print(3 % 5)

# 3和5的幂乘
print(3 ** 5)


8
-2
15
0.6
3
243


**程序清单4.10：参考答案**

## ●4.2 变量


### ○4.2.1 变量的定义

In [5]:
n = "汪星人"
print(n)

汪星人


**程序清单4.11：变量的示例①**

In [ ]:
# 如果使用print作为变量名，在调用print()函数时就会报错。
print = "Hello"
print(print) # TypeError: 'str' object is not callable

**程序清单4.12：变量的示例②**

#### □习题

In [ ]:
# 请把“喵星人”代入变量n


# 输出变量n。


# 请输出“n”这一字符串。


# 请将 3+7 这一算式代入变量 n 中


# 请对变量 n 进行输出


**程序清单4.13：习题**

#### 参考答案

In [6]:
# 请将“喵星人”代入变量 n 中
n = "喵星人"

# 请对变量 n 进行输出
print(n)

# 请输出“n”这一字符串。
print("n")

# 请将 3+7 这一算式代入变量 n 中
n = 3 + 7

# 请对变量 n 进行输出
print(n)

喵星人
n
10


**程序清单4.14：参考答案**

### 〇4.2.2 变量的更新

In [7]:
x = 1
print(x)  
x = x + 1
print(x)  

1
2


**程序清单4.15：变量更新的示例①**

In [8]:
x = 5
x *= 2
print(x)  

10


**程序清单4.16：变量更新的示例②**

#### □习题

In [ ]:
m = "喵星人"
print(m)

# 请用“汪星人”覆盖变量 m，并进行输出




n = 14
print(n)

# 请将变量 n 乘以 5，并用结果对其进行覆盖


print(n)

**程序清单4.17：习题**

#### □参考答案

In [1]:
m = "喵星人"
print(m)

# 请用“汪星人”覆盖变量 m，并进行输出
m = "汪星人"
print(m)


n = 14
print(n)

# 请将变量n乘以5，并用结果对齐进行覆盖。n = n * 5、n = 5 * n 都是正确答案
n *= 5

print(n)

喵星人
汪星人
14
70


**程序清单4.18：参考答案**

### 〇4.2.3 字符串的连接

In [10]:
m = "太郎"
print("我的名字是" + m + "哟") 

我的名字是太郎哟


**程序清单4.19：字符串连接的示例**

#### □习题

In [ ]:
# 请将“北京”代入变量p中


# 使用变量 p 和 print() 函数输出“我是北京人”文字信息



**程序清单4.20：习题**

 #### □参考答案

In [2]:
# 请将“北京”代入变量p中
p = "北京"

# 使用变量 p 和 print() 函数输出“我是北京人”文字信息
print("我是" + p + "人")

我是北京人


**程序清单4.21：参考答案**

## ●4.3 型

### ○4.3.1 类型

In [ ]:
height = 177
print("我的身高是" + height + "cm。")   # 会发生「TypeError: must be str, not int」的错误输出。

**程序清单 4.22：类型的示例①**

In [1]:
height = 177
type(height)  # 可以看到变量是int类型

int

**程序清单 4.23：类型的示例②**

#### □习题

In [ ]:
h = 1.7
w = 60

# 请将变量 h 和 w 的类型进行输出



# 请将计算结果代入到变量 bmi 中


# 请将变量 bmi 进行输出


# 请将变量 bmi 的类型进行输出


**程序清单4.24：习题**

#### □参考答案

In [13]:
h = 1.7
w = 60

# 请将变量 h 和 w 的类型进行输出
print(type(h))
print(type(w))

# 请将计算结果代入到变量 bmi 中
bmi = w / h ** 2

# 请将变量 bmi 进行输出
print(bmi)

# 请将变量 bmi 的类型进行输出
print(type(bmi))

<class 'float'>
<class 'int'>
20.761245674740486
<class 'float'>


**程序清单4.25：参考答案**

 ### ○4.3.2 类型的变换

In [17]:
h = 177
print("我的身高是" + str(h) + "cm。")

我的身高是177cm


**程序清单4.26：类型变换的示例①**

In [18]:
a = 35.4
b = 10
print(a + b)

45.4


**程序清单4.27：类型变换的示例②**

#### □习题

In [ ]:
h = 1.7
w = 60
bmi = w / h ** 2

#  请输出“您的 bmi 值为______哟。”这一信息
print("您的bmi值为" + bmi + "哟。")

**程序清单 4.28：习题**

#### □参考答案

In [14]:
h = 1.7
w = 60
bmi = w / h ** 2

# 请输出“您的 bmi 值为______哟。”这一信息
print("您的bmi值为" + str(bmi) + "哟。")

您的bmi值为20.761245674740486哟。


**程序清单4.29：参考答案**

### ○4.3.3 对类型的理解与确认

In [15]:
greeting = "嗨!"
print(greeting*2)  

嗨!嗨!


**程序清单4.30：对类型的理解和确认的示例**

#### □习题

In [19]:
n = "10"
print(n*3)

101010


**程序清单 4.31：习题**

- 是int类型，结果为30
- 是int类型，结果为101010
- 是str类型，结果为30
- 是str类型，结果为101010

#### □参考答案

是str类型，结果为101010

## ●4.4 if语句

### ○4.4.1 比较运算符

In [19]:
print(1 + 1 == 3)  

False


**程序清单4.32：比较运算符的示例**

#### □习题

In [26]:
# 请使用“!=”编写 4 + 6 与 -10 的关系式，并将 True 作为结果进行输出



**程序清单4.33：习题**

#### □参考答案

In [20]:
# 请使用“!=”编写 4 + 6 与 -10 的关系式，并将 True 作为结果进行输出
print(4 + 6 != -10)

True


**程序清单4.34：参考答案**

### ○4.4.2 if语句

In [22]:
n = 2
if n == 2:
    print("好遗憾！您是第" + str(n) + "个到达的")  # 只有当n等于2时才显示出来

好遗憾！您是第 2 个到达的


**程序清单4.35：if语句的示例①**

In [20]:
animal = "cat"
if animal == "cat":
    print("猫咪真的是好可爱啊")  # 只在animal是cat的时候显示

猫咪真的是好可爱啊


**程序清单 4.36：if语句的示例②**

#### □习题

In [30]:
n = 16

# 请使用if语句实现当变量n>15时，输出“这是一个非常大的数字”的文本信息


**程序清单 4.37：习题**

#### □参考答案

In [23]:
n = 16

# 请使用if语句实现当变量n>15时，输出“这是一个非常大的数字”的文本信息
if n > 15:
    print("这是一个非常大的数字")

这是一个非常大的数字


**程序清单4.38：参考答案**

### ○4.4.3 else

In [27]:
n = 2
if n == 1:
    print("恭喜您获胜！")  # 只有当n为1时才会被显示出来
else:
    print("好遗憾！您是第" + str(n) + "个到达的")  # 当n不是1时才会被显示出来

好遗憾！您是第 2 个到达的


**程序清单4.39：else的示例①**

In [25]:
animal = "cat"
if animal == "cat":
    print("猫咪真的是好可爱啊")  # 只有当animal为cat时才会被显示出来else:
    print("这不是猫咪呀")  # 当animal不是cat时才会被显示出来

猫咪真的是好可爱啊


**程序清单4.40：else的示例②**

#### □习题

In [ ]:
n = 14


if n > 15:
    print("这是一个非常大的数字")
# 请使用 else 输出“这是一个很小的数字”这一信息


**程序清单4.41：习题**

#### □参考答案

In [8]:
n = 14


if n > 15:
    print("这是一个非常大的数字")
# 请使用 else 输出“这是一个很小的数字”这一信息
else:
    print("这是一个很小的数字")

这是一个很小的数字


**程序清单4.42：参考答案**

### ○4.4.4 elif

In [9]:
number = 2
if number == 1:
    print("金牌！")
elif number == 2:
    print("银牌！")
elif number == 3:
    print("铜牌！")
else:
    print("很遗憾！您是第" + str(number) + "位到达终点的选手") 

银牌！


**程序清单4.43：elif の例①**

In [10]:
animal = "cat"
if animal == "cat":
    print("猫咪真的是好可爱啊")
elif animal == "dog":
    print("狗狗很酷呢")
elif animal == "elephant":
    print("大象好大个子哇")
else:
    print("不是猫咪不是狗狗也不是大象呀")

ねこはかわいいですな


**程序清单4.44：elif の例②**

#### □习题

In [ ]:
n = 14

if n > 15:
    print("这是一个非常大的数字")
# 请使用 elif 语句实现当 n 大于 11 且小于 15 时，输出“这是一个不大不小的数字”


    
else:
    print("这是一个很小的数字")

**程序清单4.45：习题**

#### □参考答案

In [29]:
n = 14

if n > 15:
    print("这是一个非常大的数字")
# 请使用 elif 语句实现当 n 大于 11 且小于 15 时，输出“这是一个不大不小的数字”
elif n >= 11:
    print("这是一个不大不小的数字")
else:
    print("这是一个很小的数字")

这是一个不大不小的数字


**程序清单4.46：参考答案**

### ○4.4.5 and・not・or

#### □习题

In [38]:
n_1 = 14
n_2 = 28

# 请将 “n_1比8大，且比14小” 编写成条件表达式，并用and进行输出
print()

# 请将 “n_1的平方比n_2的5倍小” 编写成表达式，并用not进行输出
print()


**程序清单4.47：习题**

#### □参考答案

In [28]:
n_1 = 14
n_2 = 28

# 请将 “n_1比8大，且比14小” 编写成条件表达式，并用and进行输出
print(n_1 > 8 and n_1 < 14)

# 请将 “n_1的平方比n_2的5倍小” 编写成表达式，并用not进行输出
print(not n_1 ** 2 < n_2 * 5)

False
True


**程序清单4.48：参考答案**

## ●4.5 附加习题

#### □习题

In [ ]:
# 请将阳历年份输入变量 year 中





# 使用 if 语句进行条件分支，判断某个年份是闰年还是平年


**程序清单4.49：习题**

#### □参考答案

In [32]:
# 请将阳历年份输入变量 year 中
year = 2000

# 使用 if 语句进行条件分支，判断某个年份是闰年还是平年
if year % 100 == 0 and year % 400 != 0:
    print (str(year) + "年是平年")
elif year % 4 == 0:
    print (str(year) + "年是闰年")
else:
    print (str(year) + "年是平年")

2000年是闰年


**程序清单4.50：参考答案**